### PR TITLE
Add convolutional approach

### PR DESCRIPTION
**Pull request by sarajcev**
Add convolutional network with LSTM layers for multivariate multi-step predictions. Datasets processed into shape: samples, timesteps, features.

### CODE REVIEW

File: data_log.py [hunk 1]
```diff
@@ -486,6 +527,254 @@ def mean_absolute_percentage_error(y_true, y_pred):
...
+    
+    return mean_std_values, X_train, y_train, X_test, y_test
+
+
+# In[ ]:
+
+
+# Day-ahead (i.e. 24-hours ahead) short-term load prediction with time-series data
+START_DATE = '2014-01-09'  # starting date of the time-series data
+HISTORY_SIZE = 7*24        # window size in hours for the past history values
```
Comment (sarajcev) [inline data_log.py]: Using seven-day past history window.

File: data_log.py [hunk 2]
```diff
@@ -486,6 +527,254 @@ def mean_absolute_percentage_error(y_true, y_pred):
...
+print('Training data shape:')
+print(X_train.shape, y_train.shape)
+print('Validation/Testing data shape:')
+print(X_test.shape, y_test.shape)
+
+
+# In[ ]:
+
+
+BATCH_SIZE = 256       # batch size
```
Comment (sarajcev) [inline data_log.py:737]: Batch size and number of epochs can be optimized along with network architecture.

File: data_log.py [hunk 3]
```diff
@@ -486,6 +527,254 @@ def mean_absolute_percentage_error(y_true, y_pred):
...
+
+# #### Convolutional (LSTM) deep ANN using functional `tf.keras` API
+
+# In[ ]:
+
+
+# Convolutional two-layer deep ANN without regularization
+# tf.keras functional API
+input_layer = keras.layers.Input(shape=(X_train.shape[1:]), batch_size=BATCH_SIZE)
+x = keras.layers.LSTM(units=32, return_sequences=True, activation='relu')(input_layer)
```
Comment (sarajcev) [inline data_log.py:769]: Size of the layer can be optimized. LSTM layers have regularization options which can be activated.

**Comment by sarajcev** [comment]
This can be seen as a "good first issue" of the CNN approach to the multivariate multi-step forecasting, although accuracy is not good at this point. Hyper-parameters of the CNN should be optimized, including the size (and even type) of the layers, number of layers (i.e. network depth), regularization, batch size, no. of epochs, etc. Size of the past history in creating windows can be changed as well to see how it influences the prediction accuracy.

**Comment by sarajcev** [comment]
Added a file `data_log_colab.ipynb` which is a version of the original Notebook that works with the Google Colab: https://colab.research.google.com/notebooks/intro.ipynb Using the "Colab" enables a faster model training with a GPU support on the Google's servers. Data (and model) is kept private.

**Review by NilsJakob — APPROVED** [review]
Code works, yet on one occation got mape value NaN, hence no predictive output, may be caused by neurons 'dying', because of ReLu activation function, since ReLU is 0 if x <= 0
